detect/location.go

```diff
@@ -50,6 +50,15 @@ func location(fragment Fragment, matchIndex []int) Location {
 			location.endColumn = (end - prevNewLine)
 			location.endLineIndex = newLineByteIndex
 		}
+
+		// Fixes: https://github.com/gitleaks/gitleaks/issues/1352
+		// fixes the bug that a new line from default secretSuffix
+		// leads in the reporting to a false positive in the following line
+		if location.endColumn == 1 {
+			location.endLine -= 1 // -1 because of newline in secretSuffix
+			location.endColumn = prevNewLine - location.startLineIndex
+		}
```

Comment:
Is it possible to add a test for this? Even with the comment, the reason why this is necessary isn't immediately obvious.

Reply:
I will try to create a test for this during this week.

Reply:
I have added a test case for this issue. In addition it was necessary to adjust the generic.toml within the testdata directory and to change the viper config in the test function.

@rgmz please have a look at the latest commits and let me know if something needs to be adjusted :)